### ISSUE TITLE
Configure linen models for training

### ISSUE THREAD

**Issue by mrakitaTT** [tenstorrent/tt-xla#364]
Currently we have a way to configure flax nnx models for training, but for linen models there is a TODO in the code:
```
    # TODO find another way to do this since model.train() does not exist, maybe
    # by passing train param as kwarg to __call__.
```

Flax models from the HF transformers library have the `train` parameter in the `call` function so we can just set it to true for training mode.

We also missed to add this in a bunch of testers:
```
    # @ override
    def _get_static_argnames(self):
        return ["train"]
```

Maybe the best option would be to change the `_configure_model_for_training` to update the workload and add the `"train"` to the static argnames and `"train": True` to the forward method kwargs, so we don't have to change all the testers.
